pkg/binary/envoy/fetch_test.go

```diff
@@ -15,31 +15,163 @@
 package envoy
 
 import (
+	"bytes"
+	"compress/gzip"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"os"
 	"path/filepath"
 	"testing"
 
-	"github.com/mholt/archiver/v3"
 	"github.com/stretchr/testify/require"
+	"github.com/ulikunitz/xz"
 
-	reference "github.com/tetratelabs/getenvoy/pkg"
 	"github.com/tetratelabs/getenvoy/pkg/globals"
-	"github.com/tetratelabs/getenvoy/pkg/manifest"
+	tar "github.com/tetratelabs/getenvoy/pkg/internal"
 	manifesttest "github.com/tetratelabs/getenvoy/pkg/test/manifest"
 	"github.com/tetratelabs/getenvoy/pkg/test/morerequire"
 )
 
 const envoyVersion = "1.17.2" // This is only for unit testing: we don't need to use latest.
 
+func TestUntarEnvoyError(t *testing.T) {
```

Comment:
we never tested what was in the manifest specifically before. Notably, we have "gz" and "xz" files, and they can be empty on error.